Dockerfile

```diff
@@ -13,35 +13,62 @@ COPY \
 # /opt/mlserver/dist folder
 RUN ./hack/build-wheels.sh /opt/mlserver/dist
 
-FROM python:3.8-slim
+FROM registry.access.redhat.com/ubi9/ubi-minimal
 SHELL ["/bin/bash", "-c"]
 
+ARG PYTHON_VERSION=3.8.13
+ARG CONDA_VERSION=4.13.0
+ARG MINIFORGE_VERSION=${CONDA_VERSION}-1
 ARG RUNTIMES="all"
 
 ENV MLSERVER_MODELS_DIR=/mnt/models \
     MLSERVER_ENV_TARBALL=/mnt/models/environment.tar.gz \
-    PATH=/opt/mlserver/.local/bin:$PATH
+    MLSERVER_PATH=/opt/mlserver \
+    CONDA_PATH=/opt/conda \
+    PATH=/opt/mlserver/.local/bin:/opt/conda/bin:$PATH
+
+# Install some base dependencies required for some libraries
+RUN microdnf update -y && \
+    microdnf install -y \
+        tar \
+        gzip \
+        libgomp \
+        mesa-libGL \
+        glib2-devel \
+        shadow-utils
 
-RUN apt-get update && \
-    apt-get -y --no-install-recommends install \
-        unattended-upgrades \
-        libgomp1 libgl1-mesa-dev libglib2.0-0 build-essential ffmpeg && \
-    unattended-upgrades
+# Install Conda, Python 3.8 and FFmpeg
+RUN microdnf install -y wget && \
+    wget "https://github.com/conda-forge/miniforge/releases/download/${MINIFORGE_VERSION}/Miniforge3-${MINIFORGE_VERSION}-Linux-x86_64.sh" \
+        -O miniforge3.sh && \
+    bash "./miniforge3.sh" -b -p $CONDA_PATH && \
+    rm ./miniforge3.sh && \
+    echo $PATH && \
+    conda install --yes \
+        conda=$CONDA_VERSION \
+        python=$PYTHON_VERSION \
+        ffmpeg && \
+    conda clean -tipy && \
+    microdnf remove -y wget && \
+    echo "conda activate base" >> "$CONDA_PATH/etc/profile.d/conda.sh" && \
+    ln -s "$CONDA_PATH/etc/profile.d/conda.sh" /etc/profile.d/conda.sh && \
+    echo ". $CONDA_PATH/etc/profile.d/conda.sh" >> ~/.bashrc
 
-RUN mkdir /opt/mlserver
+RUN mkdir $MLSERVER_PATH 
 WORKDIR /opt/mlserver
 
 # Create user and fix permissions
 # NOTE: We need to make /opt/mlserver world-writable so that the image is
 # compatible with random UIDs.
-RUN useradd -u 1000 -s /bin/bash mlserver -d /opt/mlserver && \
-    chown -R 1000:0 /opt/mlserver && \
-    chmod -R 776 /opt/mlserver
+RUN useradd -u 1000 -s /bin/bash mlserver -d $MLSERVER_PATH && \
```

Comment:
hope we will not get any issues when OpenShift will be running the images using random user

Reply:
I tried it out manually, and it seems to be OK. The tests also run the images with random user IDs to verify just this.

Reply:
Awesome, good to know!